pynicotine/pluginsystem.py

```diff
@@ -440,9 +440,13 @@ def toggle_plugin(self, plugin_name):
         enabled = plugin_name in self.enabled_plugins
 
         if enabled:
-            self.disable_plugin(plugin_name)
+            if self.disable_plugin(plugin_name):
+                config.sections["plugins"]["enabled"].remove(plugin_name)
```

Comment:
I feel like this should go into the enable/disable_plugin methods instead, perhaps with a `permanent` boolean parameter that's false when loading plugins on startup and disabling them on shutdown.

Reply:
> when loading plugins on startup and ~disabling~ [unloading] them on shutdown

It seems that this should be the intention of having the seperate load/unload methods, but instead these methods have become dependent upon eachother such that a refactor would be required.

> `permanent/persistent`

`remember`?

Reply:
I don't remember if we used to have load/unload methods, it's been too long.

> it would be better to avoid having to do this if possible.

We need something similar for the preferences dialog as well. It's currently possible to enable/disable plugins, and end up in an inconsistent state by pressing Cancel instead of OK. We need to write the list of enabled plugins immediately.

Reply:
Yeh I was looking at that too. The separate tracking that goes on in Preferences might as well be removed, as Plugins is one category that immediate config changes is unavoidable (yet that behaviour is desirable in most cases imo).

Reply:
Addressed in 5408c40ffa67364ef480c538929bbdff1bfc19d5

Reply:
Is there any need for the "Enable plugins" toggle? Not sure in what situation you would want to disable all plugins.

Reply:
Yes, there might be certain situations where the configured plugins are not wanted perhaps just for one session.

The advantage of having the global plugin toggle is that the user's selected combination of plugins are remembered.